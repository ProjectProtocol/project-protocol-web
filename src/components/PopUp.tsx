import { useTranslation } from 'react-i18next'
import { Modal, ModalProps } from 'react-bootstrap'
import defaultIcon from 'src/images/icon.svg'

export interface IPopUp extends ModalProps {
  title?: string
  titleHelper?: string
  icon?: string
  bodyClass?: string
}

export default function PopUp({
  bodyClass,
  children,
  title,
  titleHelper,
  icon,
  closeButton,
  ...props
}: IPopUp) {
  const { t } = useTranslation()

  return (
    <Modal centered {...props}>
      <Modal.Header closeButton={closeButton} />
      <Modal.Body className={bodyClass}>
        {title && (
          <div className="mb-3">
            <h3>
              <img
                src={icon || defaultIcon}
                alt={t('ui.ppLogoAlt')}
                className="me-2"
                style={{ height: '1.125rem' }}
              />
              {title}
            </h3>
            {titleHelper && <p className="small">{titleHelper}</p>}
          </div>
        )}
        {children}
      </Modal.Body>
    </Modal>
  )
}
