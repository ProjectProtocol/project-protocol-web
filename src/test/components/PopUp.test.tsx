import { render, screen } from '@testing-library/react'
import PopUp from 'src/components/PopUp'
import { it, expect } from 'vitest'

const defaultProps = {
  title: 'Test Popup',
  show: true,
}

const renderPopUp = (props = {}) => {
  const mergeProps = { ...defaultProps, ...props }
  return render(<PopUp {...mergeProps} />)
}

it('matches the snapshot', async () => {
  const { baseElement } = renderPopUp()
  expect(baseElement).toMatchSnapshot()
})

it('does not have a close button by default', async () => {
  const { baseElement } = renderPopUp()
  expect(baseElement.querySelector('.btn-close')).toBeFalsy()
})

it('renders a close button if specified', async () => {
  const { baseElement } = renderPopUp({ closeButton: true })
  expect(baseElement.querySelector('.btn-close')).toBeTruthy()
})

it('renders the title', async () => {
  render(<PopUp title="Test Popup" show={true} />)
  expect(screen.getByRole('heading').textContent).toBe('Test Popup')
})

it('does not render title helper if empty', () => {
  const { baseElement } = renderPopUp()
  expect(baseElement.querySelector('p')).toBeFalsy()
})

it('renders title helper text if given', async () => {
  const { baseElement } = renderPopUp({ titleHelper: 'I am additional text' })
  expect(baseElement.querySelector('p')?.textContent).toEqual(
    'I am additional text',
  )
})
