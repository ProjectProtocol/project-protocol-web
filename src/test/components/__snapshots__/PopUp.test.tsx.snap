// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
<body
  class="modal-open"
  data-rr-ui-modal-open=""
  style="padding-right: 1024px;"
>
  <div />
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal show"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        />
        <div
          class="modal-body"
        >
          <div
            class="mb-3"
          >
            <h3>
              <img
                alt="Project protocol logo"
                class="me-2"
                src="/src/images/icon.svg"
                style="height: 1.125rem;"
              />
              Test Popup
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
